src/types/register_request.rs

```diff
@@ -18,22 +18,36 @@ pub enum RegisterRequest {
     Image(ImageSpec),
 }
 
-/// Parameters of registered RTP input stream.
+/// Parameters of registered input stream.
 /// Before using input in video composition or output mixing,
-/// input has to be firstly registered using `register_input` request.
-///
-/// At least one of `video` and `audio` has to be defined.
+/// input has to be registered using the `register_input` request.
 #[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
 #[serde(deny_unknown_fields)]
-pub struct RegisterInputRequest {
-    /// An identifier for the input stream.
-    pub input_id: InputId,
-    /// UDP port or port range on which the compositor should listen for the stream.
-    pub port: Port,
-    /// Parameters of a video source included in the RTP stream.
-    pub video: Option<Video>,
-    /// Parameters of an audio source included in the RTP stream.
-    pub audio: Option<Audio>,
+#[serde(tag = "input_type", rename_all = "snake_case")]
+pub enum RegisterInputRequest {
```

Comment:
We already have too many `_type` fields there. Create 2 separate types and use entity_type field e.g. `rtp_input_stream`, `mp4`

I don't think it's necessary to have docs for new features ready in this PR, but if you follow my suggestion, you will need to update the docs for RTP.